lib/data/wrapper.js

```diff
@@ -285,6 +288,173 @@ const data = {
         }
         return client.getDiskUsage(log.getSerializedUids(), cb);
     },
+
+    /**
+     * _retryDelete - Attempt to delete key again if it failed previously
+     * @param {object} request - request object
+     * @param {string} sourceLocationConstraintType -
+     * source locationContraint type (azure, aws_s3, ...)
+     * @param {string} sourceLocationConstraintName -
+     * source locationContraint name (aws-test, azuretest, ...)
+     * @param {object} storeMetadataParams - metadata information of the
+     * source object
+     * @param {array} dataLocator - source object metadata location(s)
+     * NOTE: for Azure and AWS data backend this array
+     * @param {object} dataStoreContext - information of the
+     * destination object
+     * dataStoreContext.bucketName: destination bucket name,
+     * dataStoreContext.owner: owner,
+     * dataStoreContext.namespace: request namespace,
+     * dataStoreContext.objectKey: destination object key name,
+     * @param {BackendInfo} destBackendInfo - Instance of BackendInfo:
+     * Represents the info necessary to evaluate which data backend to use
+     * on a data put call.
+     * @param {object} serverSideEncryption - serverSideEncryption
+     * @param {object} log - Werelogs request logger
+     * @param {function} cb - callback
+     * @returns {function} cb - callback
+     */
+    copyObject: (request, sourceLocationConstraintType,
+      sourceLocationConstraintName, storeMetadataParams, dataLocator,
+      dataStoreContext, destBackendInfo, serverSideEncryption, log, cb) => {
+        // NOTE: using copyObject only if copying object from one external
+        // backend to the same external backend
```

Comment:
and for Azure if it is the same account since Azure copy outside of an account is async